packages/c-pin-input/tests/c-pin-input.cy.tsx

```diff
@@ -0,0 +1,79 @@
+/// <reference types="../../../cypress/support" />
+
+import { h } from "vue"
+import * as Examples from "../examples"
+import { CPinInput, CPinInputClearButton, CPinInputField } from "../src"
+
+const render = (props?: any) => {
+  cy.mount(() => {
+    return h(() => (
+      <CPinInput {...props}>
+        <CPinInputField />
+        <CPinInputField />
+        <CPinInputField />
+        {props?.clear && (
+          <CPinInputClearButton>Clear</CPinInputClearButton>
+        )}{" "}
+      </CPinInput>
+    ))
+  })
+}
+
+describe("CPinInput", () => {
+  Object.entries(Examples).map(([name, example]) => {
+    it(`renders ${name} successfully`, () => {
+      cy.mount(example.default).checkA11y()
+    })
+  })
+
+  it("renders default value correctly", () => {
+    const value = ["1", "2"]
+
+    render({ value })
+
+    cy.get(`input[id="pin-input:0:0"]`).should("have.value", "1")
+    cy.get(`input[id="pin-input:0:1"]`).should("have.value", "2")
+    cy.get(`input[id="pin-input:0:2"]`).should("have.value", "")
+  })
+
+  it("render with different placeholder", () => {
+    const placeholder = "X"
+    render({ placeholder })
+
+    cy.get(`input[id="pin-input:0:0"][placeholder="X"]`)
+  })
+
+  it("renders with password type", () => {
+    const value = ["1", "2"]
+
+    render({ value, mask: true })
+    cy.get(`input[id="pin-input:0:0"][type="password"]`)
+  })
+
+  it("should write correctly", () => {
+    render()
+
+    // We need to invoke val instead of use the "type" function
+    // Because Zag prevents some natural behavior of the input to replace it.
+    // So, invoking "val" is a way to type the value.
+    // The downside of this approach is that it won't naturally trigger the focus on the next element
```

Comment:
I see. Is there any way we can test the public API of the `CInput` without manually mutating the `input` value with `jest` instead of `cypress`?

The reason for that is that it allows us to focus on the public API. Mutating the `<c-pin-input />` should update the `<c-pin-input-field/>` kind of disannuls the purpose of the test. Maybe jest could handle this, and we can skip this in cypress.

Reply:
Hi @codebender828 

Thanks again for your review ! 
We can indeed test the `CInput`alone, whether it's in `jest` or `cypress`. 
However, when it comes to the `CPinInput`, it's the same in `jest` or `cypress`... I'll try one last thing in `jest` to confirm.

I think best we can do is test that the `CInput` is working great from our side and leave the rest to Zag i think...